napari/utils/io.py

```diff
@@ -31,6 +32,19 @@ def imsave(filename: str, data: np.ndarray):
         imageio.imsave(filename, data)
 
 
+def imsave_extensions():
+    """Valid extensions of files that imsave can write to.
+
+    Returns
+    ----------
+    tuple
+        Valid extensions of files that imsave can write to.
+    """
+    import imageio
+
+    return tuple(set(x for f in imageio.formats for x in f.extensions))
```

Comment:
Are we sure this returns all valid writing extensions? I see .foobar and .nonexistentext in there... LOL

Reply:
i think those are intended more as exclusion criteria, not inclusion.  The vast majority of those extensions will also not work unless the data is appropriate, and sometimes also require additional imageio plugins to be installed.  Not sure how we should handle that though

Reply:
seems like having this is better than nothing, even if not ideal

Reply:
I think imageio provides a bunch of writers classes. We should try to find the extensions for which such a class exists. Also, imho we should use tifffile directly for writing tiffs.

Reply:
We are using tifffile directly for tiffs

Reply:
We are using tifffile directly for tiffs

Reply:
changed to a more realistic set of extensions, (by testing them all with a variety of image shapes/formats).  with a TODO to write a proper plugin